Aeb pre jor (#19)
lsst/rtn-005: body.tex

\section{Introduction}

This document provides a guide to the \VRO\ approach to work management and annual planning.

> See the operations proposal \citeds{RDO-018} for a description of the full scope and high level goals of the program.

Author: ```suggestion
See the operations proposal RDO-018 for a description of the full scope and high level goals of the program.
```

\subsection{Activity Types}
\label{sec:loe}

There are two types of activities (or epics) that are planned:  activities that result in a deliverable, and level of effort (or support) activities.
Progress can be tracked on activities with a deliverable by computing the fraction of the effort that is complete (percent complete) based on completed stories linked to the epic.
Progress on \gls{LOE} work is assumed to progress proportionally with the passage of time.

\gls{LOE} activities include attending meetings, reporting on milestones, or taking part in other activities which do not directly map to a deliverable or product.
This may be particularly the case for technical managers or others in leadership roles.
In general, we strive to minimize the fraction of effort which is devoted to \gls{LOE} activities and favor those which are more directly accountable.
However, in certain cases such as operations and maintenance of telescope and facility systems, pipelines or other systems, \gls{LOE} is perfectly acceptable.
Technical staff in Chile at the summit facility may spend a much more significant fraction of time as \gls{LOE}.

> As an example, a first-order estimate is that developers will spent 30\% of their time on \gls{LOE} type activities, and the remaining 70\% of their effort is planned and tracked against well-defined deliverables.

Reviewer: That seems very optimistic for operations where most work is emergent - fix this major issue. Its not planned. Who came up with this 70/30 ?   That is more like construction.

Author: Yes, the 70/30 is a carry over assumption from Construction. What do you suggest we include here for DP and SF folks?

Reviewer: 50 50 but we can update later

\end{description}

Some of these milestones are exposed to external reviewers (usually reserved for Level 1 milestones): it is important that these be delivered on time and to specification.
Level 1 and 2 milestones are under change control once they are defined and described in a \gls{JIRA} Milestone issue. Note the change control process is under development as a Pre-Operations activity. 

Level 3 milestones are defined for use within Departments and not required to go under project change control, but properly adhering to the plan is important: your colleagues in other teams will use these milestones to align their schedules with yours, so they rely on you to be accurate.

Epics should work to achieve milestones, i.e., they may be blocking issues on the milestones.
When a detailed description of work for a given \gls{epic} is known, it is described in \gls{JIRA}.
It should then be assigned to the appropriate \glspl{cycle}.

Progress is tracked toward achieving milestones in Smartsheet by monitoring completed story points on linked issues in \gls{JIRA} epics and rolling up the total progress.

> To ensure success, \gls{JIRA} epics must be completely detailed out prior to a full 6-month cycle and total effort should be estimated out for an entire fiscal year of effort, as detailed below.

Reviewer: and all milestones should appear in Jira which is the source of truth.

Author: ```suggestion
To ensure success, \gls{JIRA} epics must be completely detailed out prior to a full 6-month cycle and total effort should be estimated out for an entire fiscal year of effort, as detailed below.
All milestones should appear in \gls{JIRA} with a milestone issue type as the source of truth. 
```

\item The \texttt{Component} field set to the appropriate Department;
\item The \texttt{Story Points} field set to a (non-zero) estimate of the effort required to complete the \gls{epic} in terms of \glspl{SP} (see \S\ref{sec:effort}).
\item The \texttt{RO Milestone ID} field set to the appropriate L3 or L2 milestone that the epic will contribute to achieving (or achieve in its totality).
  The exact \texttt{RO Milestone ID} can be found in the milestone issue or Smartsheet.
\item The \texttt{Due Date} field set to the appropriate date, which does not exceed the due date of the milestone it is labeled to achieve.
\item The \texttt{label} field is set to identify the fiscal year during which the work will be done. Examples are FY22 or FY23.

\end{itemize}
The fields above are required to have values entered because they define the connection to Smartsheet where effort-tracking for the full project is done. Other fields in the epic can also be filled in as needed.

Be aware that:

\begin{itemize}

> \item An \gls{epic} may only be assigned to a single \gls{cycle}.

Reviewer: We are not really doign cycles in ops and not cleanly like this. The epics have been event driven.

Author: Do you plan to move into a regime of cycle-driven work and construction ramps down?  That was an assumption I was working from.  We at least have the annual POP cycle that we will work within.

As such, we acknowledge that they are necessary, but seek to minimize the fraction of our resources assigned to them.
If more than a relatively small fraction of the work for a \gls{cycle} is assigned to bucket \glspl{epic}, please consider whether this is really necessary and appropriate.

> % I think this is a tie to the Construction EVMS and is not needed here.

Reviewer: At some point you may like this to know which WBS we spend more on .. but indeed its like this construction for EVMS.

Each repository on GitHub has a bug tracker associated with it.
Members of the public may report issues or make requests on the GitHub trackers.
Per the \href{https://developer.lsst.io/processes/workflow.html}{Developer Workflow}, all new work must be associated with a \gls{JIRA} ticket number before it can be committed to the repository.
It is therefore the responsibility of technical managers to file a \gls{JIRA} ticket corresponding to the GitHub ticket, to keep them synchronized with relevant information, and to ensure that the GitHub ticket is closed when the issue is resolved in \gls{JIRA}.

The GitHub issue trackers are, in some sense, not a core part of our
workflow, but they are fundamental to community expectations of how they
can interact with the project. Ensure that issues reported on GitHub are
serviced promptly.

In some cases, the technical manager responsible for a given repository
is obvious, and they can be expected to take the lead on handling
tickets. Often, this is not the case: repositories regularly span team
boundaries. Work together to ensure that all tickets are handled.

\subsubsection{Emergent Work}
\label{emergent-work}

> 

Reviewer: Again, this is probably more the norm in operations.

\item (If applicable, i.e. it is not a bucket, as defined in \S\ref{sec:bucket}) its concrete deliverable has been achieved.
\end{enumerate}

> ``Bucket'' epics should be closed at the end of the time box (i.e. end of half fiscal year and end of fiscal year).

Reviewer: We may also assign buckets to a specific milestones and close them when the milestones is achieved e.g. DR1 Bucket Epic or  DP0.2 Bucket epic.

Note that it is not permitted to close an \gls{epic} without defining at least one \gls{story} within it.
Empty \glspl{epic} can never be completed.

> \subsection{Handling Bugs \& Emergent Work}

Reviewer: if this goes out perhaps we should refer to developer.io ??  https://developer.lsst.io/work/flow.html

Author: This section can be included if you want, but with a note that it is for DP and SF, not ROO or REO (at this time).

Reviewer: Refer to developer.io -- and we should agree how this is done for all..

It is required that the \texttt{Team} field be set to the appropriate team (\jira{RFC-145}).
This indicates which manager is responsible for \gls{seeing} that the work is completed successfully.
Available teams, and the associated managers, are listed in the \href{https://developer.lsst.io/processes/jira_agile.html}{Developer Guide}; generally speaking, they align with the the work breakdown structure described in \S\ref{sec:wbs}.

> Where there is uncertainty about which team should be responsible for a particular ticket, the “Data Production Management” team may be used to indicate that the \gls{DM} \gls{Project Manager} is responsible for assigning the work.

Reviewer: I guess there is no DM project manager in ops in charge of Data Production - AD of DP ?

Author: ```suggestion
Where there is uncertainty about which team should be responsible for a particular ticket, the “Data Production Management” team may be used to indicate that the \gls{AD} of  \gls{Data Production} is responsible for assigning the work.
```

It is required that the \texttt{Team} field be set to the appropriate team (\jira{RFC-145}).
This indicates which manager is responsible for \gls{seeing} that the work is completed successfully.
Available teams, and the associated managers, are listed in the \href{https://developer.lsst.io/processes/jira_agile.html}{Developer Guide}; generally speaking, they align with the the work breakdown structure described in \S\ref{sec:wbs}.

> Where there is uncertainty about which team should be responsible for a particular ticket, the “Data Production Management” team may be used to indicate that the \gls{AD} of  \gls{Data Production} is responsible for assigning the work.

Author: ```suggestion
Where there is uncertainty about which team should be responsible for a particular ticket, the “Data Production Management” team may be used to indicate that the \gls{AD} of Data Production is responsible for assigning the work.
```
